external-crates/move/crates/move-vm-config/src/runtime.rs

```diff
@@ -63,8 +67,11 @@ impl Default for VMRuntimeLimitsConfig {
 #[cfg(debug_assertions)]
 #[derive(Clone, Debug)]
 pub struct VMProfilerConfig {
+    pub enabled: bool,
     /// Base path for files
     pub base_path: std::path::PathBuf,
+    /// User configured full path override
+    pub full_path: Option<std::path::PathBuf>,
```

Comment:
Let's continue using the idiom of `None` meaning profiling is disabled and `Some(p)` meaning enabled, and push the logic of how to generate the custom name out of the Move codebase and into the CLI codebase (i.e. get rid of the separate `enabled` flag here.

Reply:
In fact, seeing as this is the `VMProfilerConfig`, I would explore whether it's possible to push that pattern further, by making `None: Option<VMProfilerConfig>` mean that the profile is disabled, and `Some(c): Option<VMProfilerConfig>` mean enabled (that way you don't need to make the `full_path` optional either.